EOS/test_throughput.py

```diff
@@ -0,0 +1,134 @@
+import hashlib
+import os
+import glob
+import traceback
+import re
+import binascii
+import logging
+from IOUtils import ReadWriteOp, Checksum
+from timer import StopWatch, Measure, Profiling
+import argparse
+import sys
+
+dictionary = {}
+extension = ".txt"
+
+def get_args():
+    parser = argparse.ArgumentParser(description='Arguments', formatter_class = argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("--num", dest="number", type=int,
+                        required=True,
+                        help='Number of files to be generated')
+    parser.add_argument( "--file_size", dest="file_size",
+                        required = True,
+                        help='Size of the files, eg - 1M FOR 1MB, 2K for 2KB')
+    args = parser.parse_args()
+    return args
+
+
+class Benchmark():
+
+    def __init__(self, number_of_files, input_size):
+        self.number_of_files = number_of_files
+        self.exit = 0
+        self.input_size = input_size
+        self.ops = ReadWriteOp()
+        self.check = Checksum()
+        #self.check_dir = self.check_directory()
+        self.write_result = self.write_test(number_of_files, input_size)
+        self.read_result = self.read_test(number_of_files)
+
+
+    def check_directory(self):
+        if not (os.path.isdir("eos")):
+            return 1
+        return 0
+
+    def write_test(self, number_of_files, input_size):
```

Comment:
_Consider similar comments for the read_test function below._

I would refactor and define and general purpose write (read) function that lives in IOUtils.py.
For write, the params are filepath and payload. Return success/failure on writing (should handle exceptions)
For read, the param is the filepath. Returns success/failure on reading and, if successful, the content of the file.

write_test (read_test) calls this generic write (read) for all the files to be written (read).

Reply:
Such write//read function can be used also by other tests, e.g., corruption.

Reply:
Done!